### ISSUE TITLE
Logs!

### ISSUE THREAD

**Issue by ryantheleach** [AllTheMods/Server-Scripts#8]
Can you guys _Please_ Log your output somewhere.

Preferably the usual forge logs, but clearly labelled, so people unfamiliar will attach those with your errors.

We have had clueless users (because this babies them, they have 0 experience in reporting logs correctly) coming in needing help.

We ask for logs, they don't give logs (because they _can't_), they didn't get help. 

Everyone's mad.

If there was logs, we could have diagnosed the issue was that there was no Sponge version available for the forge version they were using, and updated Sponge (all it was, was a version number needing to be bumped, due to overzealous version checking, (caused by Sponge not having or even knowing what their compatible versions are aside from the version tested with))

It would also be great if you could modify your script to use the latest sponge available, and rely upon mods to have a minimum forge version set correctly, Sponge has some pretty good reporting capabilities when something fails to mix into forge, that should cover any error reports that users create.

If you must, print a warning *after* a crash, saying to attempt to use a recommended version of either, and have some way of overriding the versions your script downloads.

**Comment by OrdinatorStouff** [comment]
**`Can you guys Please Log your output somewhere.`** 
I'm not sure what you mean since our scripts make a `serverstart.log` of everything it does. If you mean logs of the server after it's _started_ (after the jar executable has been called) then that logging is out of scope of these "launcher" scripts.

The sponge option is a very basic feature. It does no checks, it doesn't even download anything. If the user is going to be using sponge we assume they have enough common sense to either 1) get the correct files for their instance or 2) heed the warning in our documentation that says sponge use is use-at-your-own-risk.

**`It would also be great if you could modify your script to use the latest sponge available`**
As said above, we don't download or fetch anything it's 100% up to the end user to get the correct files and place them in the correct directories

**`and rely upon mods to have a minimum forge version set correctly`**
Not sure what you mean exactly; this project is designed to get the user to the point where the forge server is being launched. That's it. We check java is installed, we download forge installer if it's not present yet, etc, but once everything is good this is no different than just a complex "java -jar forge.jar" shortcut.

**`Sponge has some pretty good reporting capabilities when something fails to mix into forge, that should cover any error reports that users create.`**
Wherever sponge creates those reports is no different from a server launched by direct command line or by a server launched with our script. 

**`If you must, print a warning after a crash, saying to attempt to use a recommended version of either, and have some way of overriding the versions your script downloads.`**
This project was primarily designed for modpack developers. The MCVersion, Forgeversion, etc is not typically something an end user would ever need to change as whoever created the modpack should have already specified the correct forge version to use (and included the correct sponge version if it's intended to ship with Sponge) As for "overriding the versions our script downloads" the settings/cfg can be modified to specify whatever forge version is desired, and Sponge files are not fetched automatically at all.

**Comment by OrdinatorStouff** [comment]
Ok so I just saw your [PR ](https://github.com/AllTheMods/Server-Scripts/pull/7) and I will say that the log **entries** around the sponge section of the batch are somewhat lacking BUT that doesn't really change the other points/follow-up I made I don't think.

While launching w/ sponge still continues to be one of those "not really supported" features (i.e. doesn't exist at all in our bash version) I will review it and see about making it more verbose if nothing else.

**Comment by ryantheleach** [comment]
Yeah cheers, sorry I was slightly frustrated, the user I was helping probably didn't realize there was an additional log file. Thanks for the pull and responses. Sorry I was a little annoyed earlier.

**Comment by OrdinatorStouff** [comment]
No worries, we've all been there.

I'm going to leave this issue open as a todo for me to add some more log entries in the sponge section. I am not a sponge user myself and thus don't experience firsthand problems or areas of potential improvement, so if there's anything you feel would be useful to add in the scope of this serverstart script for sponge let me know, keeping in mind I don't intend to add anything super complex and still have minimal support for the option in the first place; but I am open minded and appreciate any and all feedback.

**Comment by ryantheleach** [comment]
Yeah, most of the errors people will get will be well past this point anyway.

The only thing I can suggest, is adding a small message when this script stops before the server starts, saying that they can find the Server-Scripts log at serverstart.log and to include it with the other logs on any help requests or bug reports?
